Automattic/wp-calypso
# Woo DNA: Get the parameters (plugin name, help URL) from the current URL instead of a client-side map

Note: this PR needs to be tested and merged in lockstep with the corresponding WP.com differential.

#### Changes proposed in this Pull Request

- On https://github.com/Automattic/wp-calypso/commit/c57cacdfc87dc0f46715e01725c0690929970600, make sure that whenever an user account is created on the Woo DNA flow, the `signup_flow_name` parameter sent to the server is set to `woodna:plugin-name` instead of `jetpack-connect`. That way, we can identify if an user account was created using the Woo DNA flow and tweak the e-mails in the server and other logic.
- On https://github.com/Automattic/wp-calypso/commit/515e63b869fb4bb13a4b3f0e46764254d57132dd (and https://github.com/Automattic/wp-calypso/commit/82978b7977346e45a03ad3599956593395f055ac to add documentation), I've changed the way Calypso detects if the current flow is Woo DNA. Instead of having a client'side map with all the plugins/services that implement the Woo DNA flow, the server will inject the following parameters into the current URL:
  - `woodna_service_name`
  - `woodna_help_url`

#### Testing instructions
- Apply the wp.com differential. Make sure you're correctly sandboxing wp.com.
- Start the Woo DNA connection flow from the WCPay plugin.
- Make sure the whole UI still works as expected. The name "WooCommerce Payments" should appear in the header, the `(?) Get help` link in the footer should point to [this page](https://docs.woocommerce.com/document/payments/connection/) (still non-existant), etc.
- Create a new user account during the flow. You'll receive a Woo-branded `Activate your account` e-mail instead of a Jetpack-branded one.
